lib/auth.js

```diff
@@ -48,6 +55,66 @@ Auth.prototype.bearer = function (bearer, sendImmediately) {
   }
 }
 
+
+Auth.prototype.negotiate = function(request, authHeader, sendImmediately) {
+  var self = this
+
+  if (kerberos === undefined) {
+    // Nothing to do if we don't have Kerberos
+    // or a current credential cache
+    return
+  }
+
+  self.hasAuth = true
+
+  if (!sendImmediately) {
```

Comment:
I don't really like the idea of `sendImmediately` being set to true. 

If a server's hostname isn't in a KDC's registered list you're going to get a GSSAPI error saying that the server is unknown.  Ideally we'd get the 401 response the first time and present the challenge data to generate the header, if we don't get a 401 we don't need to provide the authentication.  Not all servers use Kerberos, but we want the ability to authenticate to the ones that do.

By keeping the default to false, we can work with what each server requires.